### DISCUSSION TITLE
Image.unlock() doesn't update Image.at() results

### DISCUSSION THREAD

**Discussion by dewitters** [Kode/Kha#1436] [Q&A]
I'm a happy Kha user for many years, but now I ran into a problem that I'm not able to quickly fix or work around.

On the HTML5 target there is the following problem: when calling Image.lock() and Image.unlock() to update the image pixels, it all works fine visually. However, Image.at(x,y) still uses the old values.

My overall intention is to replace one image from a texture atlas with another one (with lock() and unlock()), but at() still needs to work on that since I use that to detect if a user clicks on a sprite or not.

I'm not well versed in WebGL (hence the use of Kha ;)), so I was hoping any of you here knows how to fix it or a workaround would also be fine. Thanks! :)

**Comment by RobDangerous** [comment]
Does https://github.com/Kode/Kha/commit/6e0e92200946423846d367c744fc561c7df81456 help?

**Reply by dewitters** [reply]
Unfortunately already tried that (great minds think alike ;)). I also tried to call createImageData() straight away to refresh it, but that also failed.

The reason that it doesn't work is when you look at createImageData(), it creates the data coming from Image.image. So my guess is that the texture is created from Image.image. After that, you change the texture using lock() and unlock(), but the Image.image itself remains the same. And since createImageData() is using image and not the texture, it's still the old one.

My knowledge on how to link textures with in-memory stuff (and textures on GPU memory) is very limited. That's why I didn't figure it out in more detail. But my guess is that somehow that new data needs to end up in Image.image, and then createImageData() needs to be refreshed somehow.

**Reply by RobDangerous** [reply]
Oho, ok. Created #1437 for it, will have a closer look at it tomorrow.

**Reply by dewitters** [reply]
Thanks man! I really appreciate your work.
